### PR TITLE
Saved insights creation flow

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Connecting "create a new insight" to the saved insights page. Also updates to how insights get edited/saved.

To test, either have "dashboard_collaboration" turned on or set to "true" in the code

Screen Shot 2021-08-27 at 10 04 07 AM

https://user-images.githubusercontent.com/25164963/131139581-bda46e56-0821-4c1c-a078-b4210b5bf274.mov

- New main nav menu item added for "exploratory" mode and insights updated to saved insights. By default the user is directed to the "Explore" page 

https://user-images.githubusercontent.com/25164963/133276721-a98e252b-5f8e-4c9a-b4f9-d0fab2b41d30.mov

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by liyiy** [comment]
For more context I believe this will not be the final form of what the insights creation flow will look like, just wondering if what we have now makes enough sense to continue its iteration? What is confusing and what can be improved about this flow? I would encourage testing the flow out by having the `3408-saved-insights` feature flag turned on along with `user.?...?.dashboard_collaboration`.

File: frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx
Comment (paolodamico) [inline frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx:64]: Context on why we're destructuring filters now? Think the API endpoint still expected the `filters` attr

File: frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx
Comment (paolodamico) [inline frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx:51]: Context on why we're no longer sending the `name`?

File: frontend/src/scenes/insights/InsightTabs/InsightTitle.tsx
Comment (paolodamico) [inline frontend/src/scenes/insights/InsightTabs/InsightTitle.tsx:6]: If we're no longer setting the title here this component no longer makes sense, let's get rid of it

File: frontend/src/scenes/saved-insights/savedInsightsLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/saved-insights/savedInsightsLogic.ts:137]: `/saved_insights` will no longer exist, right?

File: frontend/src/scenes/saved-insights/savedInsightsLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/saved-insights/savedInsightsLogic.ts:22]: Is this really the type? Perhaps `menuItems` should have the correct constant key as a separate attribute and you just pass it here too

File: frontend/src/scenes/insights/insightLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.ts:274]: Maybe better to do "Insight saved successfully!" as title and then a description that reads something like: "{insightName} has been saved as a new insight".

File: frontend/src/scenes/insights/insightLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.ts:275]: Doing this will add multiple routes to the router and then mess up the back button.

File: frontend/src/scenes/insights/insightLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.ts:168]: I wonder if this is still needed? Are we keeping the same behavior of the editing toast or not anymore?

File: frontend/src/scenes/insights/insightLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.ts:172]: Nit: Name does not communicate this is a boolean option

**Review by paolodamico — CHANGES_REQUESTED** [review]
Started adding a few comments but can't really review fully because there's something wrong with the branch, when pulling I get a can't fast forward error. Think you need to rebase from master. Didn't do it myself to avoid changing code unexpectedly.

File: frontend/src/scenes/insights/InsightTabs/InsightTitle.tsx
Comment (liyiy) [reply frontend/src/scenes/insights/InsightTabs/InsightTitle.tsx:6]: ok~ Can I get more context on what `annotationsToCreate` is? It's in all of the insight title components but I'm not 100% sure what it's doing

File: frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx
Comment (liyiy) [reply frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx:64]: you're right here, reverting this change

File: frontend/src/scenes/insights/insightLogic.ts
Comment (liyiy) [reply frontend/src/scenes/insights/insightLogic.ts]: I forgot why I had this but it's not being used so I removed it for now

File: frontend/src/scenes/saved-insights/savedInsightsLogic.ts
Comment (liyiy) [reply frontend/src/scenes/saved-insights/savedInsightsLogic.ts:136]: `/saved_insights` should still exist as it is its own menu nav tab now

**Comment by liyiy** [comment]
Updated screenshots:
Screen Shot 2021-09-10 at 1 26 42 AM
Screen Shot 2021-09-10 at 1 26 47 AM
Screen Shot 2021-09-10 at 1 27 04 AM

failing SAML test seems unrelated

File: frontend/src/scenes/funnels/funnelLogic.ts
Comment (liyiy) [inline frontend/src/scenes/funnels/funnelLogic.ts:579]: we still want to create a new insight when clicking on a new insight tab

File: frontend/src/scenes/insights/Insights.tsx
Comment (liyiy) [reply frontend/src/scenes/insights/Insights.tsx:178]: https://user-images.githubusercontent.com/25164963/133279501-3d7e952c-f9cc-4a8c-a9b2-6a28e8a55962.mov

File: frontend/src/scenes/funnels/funnelLogic.ts
Comment (mariusandra) [inline frontend/src/scenes/funnels/funnelLogic.ts:580]: This `actions.createInsight` is actually `insightHistoryLogic.actions.createInsight`, which calls `api.create('api/insight')`. The problem is that `funnelLogic` is also mounted for each dashboard item, so this makes a lot of mess when loading a dashboard:

image

You can just `if (!props.dashboardItemId)` all the code that touches `insightLogic` as a quick solution.

File: frontend/src/scenes/paths/pathsLogic.ts
Comment (mariusandra) [inline frontend/src/scenes/paths/pathsLogic.ts:132]: Paths can also be shown on dashboards

File: frontend/src/scenes/retention/retentionTableLogic.ts
Comment (mariusandra) [inline frontend/src/scenes/retention/retentionTableLogic.ts:213]: same here

File: frontend/src/scenes/trends/trendsLogic.ts
Comment (mariusandra) [inline frontend/src/scenes/trends/trendsLogic.ts:355]: I think this is also started when we're on a dashboard, so it's anybody's guess what value `insightLogic` will be pointing to.

**Review by mariusandra — CHANGES_REQUESTED** [review]
Found a few conflicts with dashboard logic, otherwise looks really good and can't wait to get this in! :)

**Review by paolodamico — CHANGES_REQUESTED** [review]
Product-based review, as @mariusandra has reviewed the code. As a preamble, this is great so far!

### No feature flag
1. When creating a new insight, this is the view you see. I found three things confusing: a) the "Insight #{x}", if this is the insight ID I think it'll cause confusion on Cloud, b) it's confusing to see "Edit" when you haven't actually saved anything, c) there's two "add to dashboard" buttons.
   
2. The funnels page still has a save button.
    

### With enabled feature flag
3. UX-wise I think it's confusing that you can have the same insight named differently when saving on a dashboard than when saving it standalone (as when saving you get asked for a new insight name)
4. Clicking New Insight -> User Paths is broken
5. I'd remove the created by filter when on the "Your insights" tab
6. Some insight icons in the type of insight filter seem confusing as we use them for other things (e.g. sessions, stickiness)
7. Currently there's no way to access the `/saved_insights` page
8. In a fresh installation I had a saved insight created as "Funnel x3" but it was actually a retention graph.
9. There's some weird positioning of elements in some rows.
Screenshot 2021-09-16 at 6 07 24 PM
10. It's not obvious that you can click the 3 dots menu, I would change the color to `$primary`.
11. The more filter box does not have the appropriate rounding and shadowing.
    
12. I would add the relevant icons to the dropdown menu actions (like we have in the dashboard page).

File: frontend/src/scenes/funnels/funnelLogic.ts
Comment (liyiy) [reply frontend/src/scenes/funnels/funnelLogic.ts]: good catch!!! I've feature flagged this (since it's a saved insights flow functionality) and also made sure to update only when an insight actually exists to be updated

**Comment by liyiy** [comment]
Addressed @mariusandra's comments and most of @paolodamico's (couldn't reproduce some of the issues). Will be leaving some of the UX improvements for a future PR so we can try to get this in :D

**Comment by paolodamico** [comment]
Makes sense @liyiy, I agree with merging and leaving UI/UX improvements for a subsequent PR. I'll just defer to @mariusandra for the final 👌 as he reviewed the code

**Review by mariusandra — CHANGES_REQUESTED** [review]
Hey, unfortunately I can't approve it. The "loadResultsSuccess" listeners are still wrong and don't explicitly take into account that none of this code should be run if we're on a dashboard. Please keep the old `if (!props.dashboardItemId) {` code and move the new logic inside that.

**Review by mariusandra — APPROVED** [review]

**Pull request merged**
